File: DME_GAP_RESOLUTION_BUNDLE_v1_FIXED/dme-platform/infra/vault/client/python/vault_client.py
from __future__ import annotations

from dataclasses import dataclass
from typing import Any, Dict

import os

import hvac  # type: ignore


@dataclass
class VaultClient:
    url: str
    token: str

    @classmethod
    def from_env(cls) -> "VaultClient":
        url = os.environ.get("VAULT_ADDR")
        token = os.environ.get("VAULT_TOKEN")
        if not url or not token:
            raise RuntimeError("VAULT_ADDR and VAULT_TOKEN must be set")
        return cls(url=url, token=token)

    def _client(self) -> hvac.Client:
        return hvac.Client(url=self.url, token=self.token)

    def read_kv2(self, path: str) -> Dict[str, Any]:
        client = self._client()
        resp = client.secrets.kv.v2.read_secret_version(path=path)
        return resp["data"]["data"]

File: DME_GAP_RESOLUTION_BUNDLE_v1_FIXED/dme-platform/infra/vault/client/nodejs/vault_client.js
import process from 'node:process';

export class VaultClient {
  constructor(url, token) {
    if (!url || !token) {
      throw new Error('VAULT_ADDR and VAULT_TOKEN must be set');
    }
    this.url = url;
    this.token = token;
  }

  static fromEnv() {
    return new VaultClient(process.env.VAULT_ADDR, process.env.VAULT_TOKEN);
  }

  async readKv2(path) {
    // Minimal example using fetch; replace with full-featured client if desired.
    const resp = await fetch(`${this.url}/v1/${path}`, {
      headers: { 'X-Vault-Token': this.token }
    });
    if (!resp.ok) {
      throw new Error(`Vault read failed: ${resp.status}`);
    }
    const data = await resp.json();
    return data.data.data;
  }
}
